broker/src/asterisk/asterisk_pbx.erl

```diff
@@ -87,12 +87,25 @@ record(AsteriskFilename, FileName, StopKeys, Timeout, {?MODULE, Pid}) ->
     file:delete(TempFile)
   end.
 
-dial(Channel, Address, undefined, {?MODULE, Pid}) ->
+monitor_session(_Pid, undefined) ->
+  undefined;
+monitor_session(Pid, AsteriskFilename) ->
+  agi_session:mix_monitor(Pid, [AsteriskFilename]).
+
+stop_monitoring(_Pid, undefined, _LocalFilename) ->
+  undefined;
+stop_monitoring(Pid, AsteriskFilename, LocalFilename) ->
+  agi_session:stop_mix_monitor(Pid),
+  file:rename(AsteriskFilename, LocalFilename).
+
+dial(Channel, Address, undefined, LocalFilename, AsteriskFilename, {?MODULE, Pid}) ->
+  monitor_session(Pid, AsteriskFilename),
```

Comment:
Was this tested when recording is not enabled? I think `LocalFilename` and/or `AsteriskFilename` can be `undefined`.

Reply:
I've tested it locally, yes. Both `monitor_session` and `stop_monitoring` have special definitions matching an `undefined` `AsteriskFilename` that perform a no-op - so we don't perform the recordings if they are not enabled.